tools/lint/src/test/kotlin/AndroidxNullabilityTests.kt

```diff
@@ -0,0 +1,130 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.lint.checks
+
+import com.android.tools.lint.checks.infrastructure.LintDetectorTest
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Issue
+
+private const val EXPECTED_ERROR = "Use androidx nullability annotations"
+private const val NO_WARNINGS = "No warnings."
+
+private fun annotationSource(pkg: String, name: String): String {
+    return """
+        package $pkg;
+
+        public @interface $name {}
+    """.trimIndent()
+}
+
+private fun javaxAnnotation(name: String): String {
+    return annotationSource("javax.annotation", name)
+}
+
+private fun androidxAnnotation(name: String): String {
+    return annotationSource("androidx.annotation", name)
+}
+
+private val NULLABLE_CLASS = """
```

Comment:
You might consider naming this `JAVAX_NULLABLE_CLASS` to contrast more directly against the `ANDROIDX` flavor, especially since the latter is actually preferred.

Reply:
done